release-notes/schema/dotnet-required-packages.json

```diff
@@ -0,0 +1,170 @@
+{
+  "description": "The set of packages required by a given .NET version for a set of distros.",
+  "type": [
+    "object",
+    "null"
+  ],
+  "properties": {
+    "ChannelVersion": {
+      "type": "string"
+    },
+    "EolDate": {
+      "type": "string",
+      "format": "date"
+    },
+    "Packages": {
+      "type": "array",
+      "items": {
+        "description": "A logical package that will exist in various distros with different package names. Includes the scenarios for which the package is required.",
+        "type": [
+          "object",
+          "null"
+        ],
+        "properties": {
+          "Id": {
+            "type": "string"
+          },
+          "Name": {
+            "type": "string"
+          },
+          "References": {
+            "type": [
+              "array",
+              "null"
+            ],
+            "items": {
+              "type": [
+                "string",
+                "null"
+              ]
+            }
+          },
+          "Required": {
+            "type": [
+              "array",
+              "null"
+            ],
+            "items": {
+              "type": [
+                "string",
+                "null"
+              ]
+            }
+          }
+        },
+        "required": [
+          "Id",
+          "Name",
+          "References",
+          "Required"
+        ]
+      }
+    },
+    "Distributions": {
+      "type": "array",
+      "items": {
+        "description": "An operating system distribution, with required package install commands and specific packages for distribution releases.",
+        "type": [
+          "object",
+          "null"
+        ],
+        "properties": {
+          "Name": {
+            "type": "string"
+          },
+          "InstallCommands": {
+            "type": [
+              "array",
+              "null"
+            ],
+            "items": {
+              "description": "A command to be run to install packages",
+              "type": [
+                "object",
+                "null"
+              ],
+              "properties": {
+                "RunUnderSudo": {
+                  "type": "boolean"
+                },
+                "CommandRoot": {
+                  "type": "string"
+                },
+                "CommandParts": {
+                  "type": "array",
+                  "items": {
+                    "type": [
+                      "string",
+                      "null"
```

Comment:
Is `"null"` necessary to allow an empty array?

Reply:
That property is defined like this: `IList<string> CommandParts` (AKA, not nullable).

What is the reason for that design choice @eiriktsarpalis?

Reply:
Fixed that with a schema update.